utilities/backupable/backupable_db.cc

```diff
@@ -2142,106 +2141,118 @@ Status BackupEngineImpl::BackupMeta::LoadFromFile(
+    } else {
+      // process the line below
     }
+  } else {
+    line.clear();
+  }
+  uint32_t num_files = UINT32_MAX;
+  if (!line.empty() || backup_meta_reader->ReadLine(&line)) {
+    num_files = static_cast<uint32_t>(strtoul(line.c_str(), nullptr, 10));
   }
-
-  num_files = static_cast<uint32_t>(strtoul(data.data(), &next, 10));
-  data.remove_prefix(next - data.data() + 1); // +1 for '\n'
-
   std::vector<std::shared_ptr<FileInfo>> files;
+  while (backup_meta_reader->ReadLine(&line)) {
+    std::vector<std::string> components = StringSplit(line, ' ');
 
-  // WART: The checksums are crc32c, not original crc32
-  Slice checksum_prefix("crc32 ");
+    if (components.size() < 1) {
+      return Status::Corruption("Empty line instead of file entry.");
+    }
 
-  for (uint32_t i = 0; s.ok() && i < num_files; ++i) {
-    auto line = GetSliceUntil(&data, '\n');
-    // filename is relative, i.e., shared/number.sst,
-    // shared_checksum/number.sst, or private/backup_id/number.sst
-    std::string filename = GetSliceUntil(&line, ' ').ToString();
+    const std::string& filename = components[0];
 
     uint64_t size;
     const std::shared_ptr<FileInfo> file_info = GetFile(filename);
     if (file_info) {
       size = file_info->size;
     } else {
       std::string abs_path = backup_dir + "/" + filename;
-      try {
-        size = abs_path_to_size.at(abs_path);
-      } catch (std::out_of_range&) {
-        return Status::Corruption("Size missing for pathname: " + abs_path);
+      auto e = abs_path_to_size.find(abs_path);
+      if (e == abs_path_to_size.end()) {
+        return Status::Corruption("Pathname in meta file not found on disk: " +
+                                  abs_path);
       }
+      size = e->second;
     }
 
-    if (line.empty()) {
+    if (components.size() < 3) {
       return Status::Corruption("File checksum is missing for " + filename +
                                 " in " + meta_filename_);
     }
 
-    uint32_t checksum_value = 0;
-    if (line.starts_with(checksum_prefix)) {
-      line.remove_prefix(checksum_prefix.size());
-      checksum_value = static_cast<uint32_t>(strtoul(line.data(), nullptr, 10));
-      if (line != ROCKSDB_NAMESPACE::ToString(checksum_value)) {
-        return Status::Corruption("Invalid checksum value for " + filename +
-                                  " in " + meta_filename_);
-      }
-    } else {
+    // WART: The checksums are crc32c, not original crc32
+    if (components[1] != "crc32") {
       return Status::Corruption("Unknown checksum type for " + filename +
                                 " in " + meta_filename_);
     }
 
+    uint32_t checksum_value =
+        static_cast<uint32_t>(strtoul(components[2].c_str(), nullptr, 10));
+    if (components[2] != ROCKSDB_NAMESPACE::ToString(checksum_value)) {
+      return Status::Corruption("Invalid checksum value for " + filename +
+                                " in " + meta_filename_);
+    }
```

Comment:
How would this happen?  I think the original code was really checking to make sure the checksum_value was at EOL.  I guess it might happen if the checksum_value had extra characters at the end? But we do not check that condition anywhere else (what if num_files=1234ABCD?)